ecosystem/python/sdk/aptos_sdk/ed25519.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Aptos
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+from nacl.signing import SigningKey, VerifyKey
+import io
+import typing
+import unittest
+
+from bcs import Deserializer, Serializer
+
+
+class PrivateKey:
+    LENGTH: int = 32
+
+    key: SigningKey
+
+    def __init__(self, key: SigningKey):
+        self.key = key
+
+    def __eq__(self, other: PrivateKey):
+        return self.key == other.key
+
+    def __str__(self):
+        return self.key.encode().hex()
+
+    def public_key(self) -> PublicKey:
+        return PublicKey(self.key.verify_key)
+
+    def random() -> PrivateKey:
+        return PrivateKey(SigningKey.generate())
+
+    def sign(self, data: bytes) -> Signature:
+        return Signature(self.key.sign(data).signature)
+
+    def deserialize(deserializer: Deserializer) -> PrivateKey:
+        key = deserializer.bytes()
+        if len(key) != PrivateKey.LENGTH:
+            raise Exception("Length mismatch")
+
+        return PrivateKey(SigningKey(key))
+
+    def serialize(self, serializer: Serializer):
+        serializer.bytes(self.key.encode())
+
+
+class PublicKey:
+    LENGTH: int = 32
+
+    key: VerifyKey
+
+    def __init__(self, key: VerifyKey):
+        self.key = key
+
+    def __eq__(self, other: PrivateKey):
+        return self.key == other.key
+
+    def __str__(self) -> str:
+        return self.key.encode().hex()
```

Comment:
Do we want to prefix these with `0x`?

Reply:
Hahaha, the standard is to go with whatever is easy ... right :)

Reply:
fixed